project_task_tags/views/project_task.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_task_form2" model="ir.ui.view">
+        <field name="name">project.task.allowed_tags.form.inherit</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_form2" />
+        <field name="arch" type="xml">
+            <field name="tag_ids" position="before">
+                <field name="allowed_tag_ids" widget="many2many_tags" invisible="1" />
+            </field>
+            <field name="tag_ids" position="attributes">
+                <attribute name="domain">[('id','in',allowed_tag_ids)]</attribute>
```

Comment:
Can you try this (assuming `project_id` is in the view)?
```suggestion
                <attribute name="domain">['|', (not project_id and 1 or 0, '=', 1), ('project_ids', 'in', project_id)</attribute>
```
if it works you can drop the rest of the changes :wink:

Reply:
yep, that seems to work fine on the view